src/libaktualizr-posix/asn1/asn1_message.cc

```diff
@@ -53,14 +54,16 @@ void SetString(OCTET_STRING_t* dest, const std::string& str) {
   OCTET_STRING_fromBuf(dest, str.c_str(), static_cast<int>(str.size()));
 }
 
-Asn1Message::Ptr Asn1Rpc(const Asn1Message::Ptr& tx, const struct sockaddr_storage& client) {
-  int socket_fd = socket(AF_INET6, SOCK_STREAM, 0);
+Asn1Message::Ptr Asn1Rpc(const Asn1Message::Ptr& tx, const struct sockaddr_in& server_sock_addr) {
+  int socket_fd = socket(AF_INET, SOCK_STREAM, 0);
   if (socket_fd < 0) {
     throw std::system_error(errno, std::system_category(), "socket");
   }
+  // what's the point in usage of SocketHandle
```

Comment:
I have no idea.

Reply:
I'll get rid of it in the next PR.

Reply:
I guess it automatically closes the socket, once it's out of scope.

Reply:
Yeah, your're right. Normally, such RAII objects includes object initialization/creation and deinitialization/deletion at ctor and dtor correspondingly. Also, the dtor's part should include shutdown and preferably verification if a socket was actually opened.